torch/csrc/autograd/engine.cpp

```diff
@@ -537,6 +541,16 @@ auto Engine::evaluate_function(NodeTask& task) -> void {
     if (!fn_info.needed_) return;
   }
 
+  // Switches to a function's CUDA stream (if applicable)
+  const auto opt_parent_stream = (*task.fn_).stream(c10::DeviceType::CUDA);
+  std::unique_ptr<c10::Event> parent_event{nullptr};
```

Comment:
Can't this just be an `c10::optional<c10::Event>`? No need for the pointer?

Reply:
Good catch. I'll fix it up post-review(s).